databus-connector/src/main/java/com/sequenceiq/cloudbreak/sigmadbus/SigmaDatabusClient.java

```diff
@@ -0,0 +1,160 @@
+import com.sequenceiq.cloudbreak.sigmadbus.model.DatabusRequestContext;
+import com.sequenceiq.cloudbreak.telemetry.databus.AbstractDatabusStreamConfiguration;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.opentracing.Tracer;
+
+public class SigmaDatabusClient<D extends AbstractDatabusStreamConfiguration> implements AutoCloseable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SigmaDatabusClient.class);
+
+    private final Tracer tracer;
+
+    private final SigmaDatabusConfig sigmaDatabusConfig;
+
+    private final D databusStreamConfiguration;
+
+    private ManagedChannelWrapper managedChannelWrapper;
+
+    public SigmaDatabusClient(Tracer tracer,
+            SigmaDatabusConfig sigmaDatabusConfig,
+            D databusStreamConfiguration) {
+        this.tracer = tracer;
+        this.sigmaDatabusConfig = sigmaDatabusConfig;
+        this.databusStreamConfiguration = databusStreamConfiguration;
+    }
+
+    /**
+     * Upload data into databus. If the payload is larger than 1 MB, the data will be uploaded to cloudera S3.
+     * @param request databus record payload input
+     * @throws DatabusRecordProcessingException error during databus record processing
+     */
+    public void putRecord(DatabusRequest request) throws DatabusRecordProcessingException {
+        ManagedChannelWrapper channelWrapper = getMessageWrapper();
+        DbusProto.PutRecordRequest recordRequest = convert(request, databusStreamConfiguration);
+        String requestId =  RequestIdUtil.newRequestId();
+        LOGGER.debug("Creating databus request with request id: {}", requestId);
+        buildMdcContext(request, requestId);
+        DbusProto.PutRecordResponse recordResponse = newStub(channelWrapper.getChannel(),
+                requestId, ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN)
+                .putRecord(recordRequest);
+        DbusProto.Record.Reply.Status status = recordResponse.getRecord().getStatus();
+        if (DbusProto.Record.Reply.Status.SENT.equals(status)) {
+            String recordId = recordResponse.getRecord().getRecordId();
+            LOGGER.debug("Dbus record sucessfully processed with record id: {}", recordId);
+        } else if (DbusProto.Record.Reply.Status.PENDING.equals(status)) {
+            String recordId = recordResponse.getRecord().getRecordId();
+            String s3BucketUrl = recordResponse.getRecord().getUploadUrl();
+            LOGGER.debug("Dbus record can be uploaded to s3 [record id: {}], [s3 url: {}]", recordId, s3BucketUrl);
+            uploadRecordToS3(s3BucketUrl, request, recordId);
+        } else {
+            throw new DatabusRecordProcessingException("Cannot process record to Sigma Databus.");
+        }
+    }
+
+    private void uploadRecordToS3(String s3Url, DatabusRequest request, String recordId) throws DatabusRecordProcessingException {
+            String payload = getPayload(request);
```

Comment:
maybe null/empty checks for the arguments just to be sure

Reply:
we can't put nulls into the blocking queue, and its getters are optional, so we are good (I know it's more robust to check that at multiple places so even unit tests can't break that but i do not like to doing that)

Reply:
imagine if we were using kotlin... :P